soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -0,0 +1,1046 @@
+                        else
+                            ImGui::Text("(%d/%d) - Vanilla", areaChecksGotten[obj.rcArea], areaChecksTotal[obj.rcArea]);
+                    } else {
+                        ImGui::Text("(%d/%d)", areaChecksGotten[obj.rcArea], areaChecksTotal[obj.rcArea]);
+                    }
+                } else {
+                    ImGui::Text("???", areaChecksGotten[obj.rcArea], areaChecksTotal[obj.rcArea]);
+                }
+
+                ImGui::PopStyleColor();
+
+                //Keep areas loaded between transitions
+                if (currentArea == obj.rcArea && doAreaScroll)
+                    ImGui::SetScrollHereY(0.0f);
+            }
+
+            areaMask <<= 1;
+        }
+
+        if (doDraw && isThisAreaSpoiled)
+            DrawLocation(obj, &checkStatusMap.find(obj.rc)->second);
+    }
+
+    //Clean up last area
+    if (doDraw)
+        ImGui::TreePop();
+
+    ImGui::EndTable(); //Checks Lead-out
+    ImGui::EndTable(); //Quick Options Lead-out
+    EndFloatWindows();
+    if (doingCollapseOrExpand) {
+        optCollapseAll = false;
+        optExpandAll = false;
+    }
+}
+
+// Windowing stuff
+void BeginFloatWindows(std::string UniqueName, ImGuiWindowFlags flags) {
+    ImGuiWindowFlags windowFlags = flags;
+
+    if (windowFlags == 0) {
+        windowFlags |=
+            ImGuiWindowFlags_AlwaysVerticalScrollbar | ImGuiWindowFlags_NoFocusOnAppearing;
+    }
+
+    if (!CVar_GetS32("gCheckTrackerWindowType", 1)) {
+        ImGui::SetNextWindowViewport(ImGui::GetMainViewport()->ID);
+        windowFlags |= ImGuiWindowFlags_NoDocking | ImGuiWindowFlags_NoNav | ImGuiWindowFlags_NoTitleBar |
+                       ImGuiWindowFlags_NoScrollWithMouse | ImGuiWindowFlags_NoScrollbar;
+
+        if (!CVar_GetS32("gCheckTrackerHudEditMode", 0)) {
+            windowFlags |= ImGuiWindowFlags_NoInputs | ImGuiWindowFlags_NoMove;
+        }
+    }
+    ImGui::PushStyleColor(ImGuiCol_WindowBg, Color_Background);
+    ImGui::PushStyleColor(ImGuiCol_Border, ImVec4(0, 0, 0, 0));
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowRounding, 4.0f);
+    ImGui::Begin(UniqueName.c_str(), nullptr, windowFlags);
+}
+void EndFloatWindows() {
+    ImGui::PopStyleVar();
+    ImGui::PopStyleColor();
+    ImGui::PopStyleColor();
+    ImGui::End();
+}
+
+uint8_t gRandomizeShopsanity;
+uint8_t gRandomizeShuffleTokens;
+uint8_t gRandomizeShuffleBeans;
+uint8_t gRandomizeShuffleScrubs;
+uint8_t gRandomizeShuffleMerchants;
+uint8_t gRandomizeShuffleCows;
+uint8_t gRandomizeShuffleAdultTrade;
+uint8_t gRandomizeShuffleKokiriSword;
+uint8_t gRandomizeShuffleWeirdEgg;
+uint8_t gRandomizeShuffleGerudoCard;
+uint8_t gRandomizeShuffleFrogSongRupees;
+uint8_t gRandomizeStartingMapsCompasses;
+uint8_t gRandomizeKeysanity;
+uint8_t gRandomizeGerudoFortressKeys;
+uint8_t gRandomizeBossKeysanity;
+uint8_t gRandomizeShuffleGanonBossKey;
+uint8_t gRandomizeGerudoFortress;
```

Comment:
not the biggest fan of these being global, maybe make them static in `IsVisibleInCheckTracker` instead?

Reply:
My native language is C#, still getting used to variable scope problems like this. In C#, these would inherently be private class variables and wouldn't be visible outside of this file. I will indeed fix up the variable scope.

Reply:
this file doesn't have a class, so while these variables would only be available in the scope of this file (unless someone tried to `#include` the `.cpp` somewhere), it still feels odd having them here

Reply:
Resolved by wrapping everything in a namespace with f34934531de18172afa3d0ba7c8db5275b048bb9.